### PR TITLE
8282036: Change java/util/zip/ZipFile/DeleteTempJar.java to stop HttpServer cleanly in case of exceptions

### PR DESCRIPTION
**Pull request by thejasviv**
This is a simple change to add some protective code to java/util/zip/ZipFile/DeleteTempJar.java so that the clean-up is more complete in the event of encountering an any exceptions while running. In the present state, the server.stop() does not get executed in case of encountering a failure. As a result, the test may get hung until a timeout. This fix should address the scenario.

The test passes on all supported platforms post fix.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8282036](https://bugs.openjdk.org/browse/JDK-8282036): Change java/util/zip/ZipFile/DeleteTempJar.java to stop HttpServer cleanly in case of exceptions

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9293/head:pull/9293` \
`$ git checkout pull/9293`

Update a local copy of the PR: \
`$ git checkout pull/9293` \
`$ git pull https://git.openjdk.org/jdk pull/9293/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9293`

View PR using the GUI difftool: \
`$ git pr show -t 9293`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9293.diff">https://git.openjdk.org/jdk/pull/9293.diff</a>

</details>

### CODE REVIEW

**Comment by jaikiran** [comment]
Hello Thejasvi, the change looks good to me. Please update the copyright year to 2022 in this changed file.

File: test/jdk/java/util/zip/ZipFile/DeleteTempJar.java [hunk 1]
```diff
@@ -72,16 +72,21 @@ public void handle(HttpExchange e) {
...
-        server.stop(0);
+            JarURLConnection c = (JarURLConnection)url.openConnection();
+            JarFile f = c.getJarFile();
+            check(f.getEntry("entry") != null);
+            System.out.println(f.getName());
+        } finally {
+            if (server != null) {
+                server.stop(0);
+            }
+        }
```
Comment (AlanBateman) [inline test/jdk/java/util/zip/ZipFile/DeleteTempJar.java:89]: Is there a reason to put the server.start in the try block, I would think this is simpler:

```
server.start();
try {
    :
} finally {
    server.stop(0);
}
```

Comment (thejasviv) [reply test/jdk/java/util/zip/ZipFile/DeleteTempJar.java:89]: Thank you @jaikiran , @AlanBateman for your reviews. I have updated the patch with your inputs.

File: test/jdk/java/util/zip/ZipFile/DeleteTempJar.java [hunk 2]
```diff
@@ -72,16 +72,21 @@ public void handle(HttpExchange e) {
...
-        JarFile f = c.getJarFile();
-        check(f.getEntry("entry") != null);
-        System.out.println(f.getName());
-        server.stop(0);
+            JarURLConnection c = (JarURLConnection)url.openConnection();
+            JarFile f = c.getJarFile();
+            check(f.getEntry("entry") != null);
+            System.out.println(f.getName());
+        } finally {
+            if (server != null) {
```
Comment (jaikiran) [inline test/jdk/java/util/zip/ZipFile/DeleteTempJar.java:86]: The null check can be removed here as suggested by Alan.

**Review by jaikiran — APPROVED** [review]
Thank you for doing those changes. The latest state of this PR looks good to me.

**Review by AlanBateman — APPROVED** [review]
Looks okay. You'll ned to edit the edit the title to align with the description of the JBS issue before you can integrate.

**Comment by thejasviv** [comment]
Thank you much @jaikiran for timely reviews, feedback, and sponsoring this commit. Also thank you @AlanBateman and @dfuch for your kind review and feedback.

**Pull request closed without merging**